test/integrations/testUtils.ts

```diff
@@ -104,6 +97,42 @@ export const overrideDestination = (destination: Destination, overrideConfigValu
   });
 };
 
+export const produceTestData = (testData: TestCaseData[]) => {
+  const result: any = [];
+  testData.forEach((tcData) => {
+    let events;
+    try {
+      switch (tcData.feature) {
+        case tags.FEATURES.PROCESSOR:
+          events = tcData.input.request.body;
+          break;
+        case tags.FEATURES.BATCH:
+          events = tcData.input.request.body.input;
+          break;
+        case tags.FEATURES.ROUTER:
+          events = tcData.input.request.body.input;
+          break;
+      }
+    } catch (e) {
+      throw new Error(`Error in producing test data: ${e}`);
+    }
+
+    events.map((event) => {
+      result.push(event.message);
```

Comment:
As we are only sending `message` without destinationConfig, test data might end up duplicating & not exactly covering all scenarios in the repository.
Not in this PR, but we should be able to extend by collecting `event.destination` & provide option in `test/scripts/testDataGenerator.ts` to take destId for updating config too via API before sending payload